File: sc381998-cs50-2018-x-sentimental-caesar/caesar.py
import cs50
import sys
def main():
    if len(sys.argv)!=2:
        print("Usage python caesar.py key")
        exit(1)
    k=int(sys.argv[1]);
    plaintext=cs50.get_string()
    for c in plaintext:
        if(c.islower()):
            print( chr((ord(c)-ord('a')+k)%26+ord('a')),end="")
        elif(c.isupper()):
            print( chr((ord(c)-ord('A')+k)%26+ord('A')), end="")
        else:
            print(c,end="")

    print()

if __name__=="__main__":
    main()

File: sc381998-cs50-2018-x-crack-20180917T043657Z/caesar.c
#include<stdio.h>
#include<cs50.h>
#include<stdlib.h>
#include<string.h>
#include<ctype.h>


int main(int argc, char** argv)
{
    if(argc!=2)
    {
        printf("Usage: ./caesar k\n");
        return 1;
    }
    int k= atoi(argv[1]);
    if(k<0)
   {
       printf("key must be positive\n");
       return 1;
   }
   string plaintext=get_string("plaintext: ");
   printf("ciphertext: ");
   for(int i=0,len=strlen(plaintext);i<len;i++)
   {
       if(islower(plaintext[i]))
       printf("%c",(plaintext[i]-'a'+k)%26+'a');
      else if(isupper(plaintext[i]))
       printf("%c",(plaintext[i]-'A'+k)%26+'A');
       else
       printf("%c",plaintext[i]);
   }
   printf("\n");
   }